SW4STM32/i2cio/Application/User/encoder.c

```diff
@@ -61,21 +61,19 @@ static inline bool detect(bool x1, bool x2, bool y1, bool y2)
 	return (x2 ^ y1) & ~(x1 ^ y2);
 }
 
-void encoderCapture(){
+void encoderCapture() {
 	for (uint8_t i = 0; i < MAX_ENCODER_COUNT; ++i){
 		if (encoders[i].enabled){
 			bool stateA = pinState(encoders[i].pinA);
 			bool stateB = pinState(encoders[i].pinB);
-			bool _stateA = encoders[i].lastStateA;
-			bool _stateB = encoders[i].lastStateB;
-			
-			if (detect(_stateA, _stateB, stateA, stateB))
-				encoders[i].unreadedValue++;
-			else if (detect(_stateB, _stateA, stateB, stateA))
-				encoders[i].unreadedValue--;
-
-			encoders[i].lastStateA = stateA;
-			encoders[i].lastStateB = stateB;
+			if(((encoders[i].stateA & 1) != (int)stateA) || ((encoders[i].stateB & 1) != (int)stateB)) { // если состояние изменилось
+				encoders[i].stateA = ((encoders[i].stateA << 1) | (int)stateA) & 0xf; // записать новое состояние в буфер состояний
+				encoders[i].stateB = ((encoders[i].stateB << 1) | (int)stateB) & 0xf; //
+				if(encoders[i].stateA == 6 && encoders[i].stateB == 3) // цифры 3 и 6 это битовые шаблоны соответствующих состояний
```

Comment:
Instead of magic 3 and 6 use binary literals (e.g. `0b00000111`)

Reply:
Done.